core/src/consensus/tendermint/mod.rs

```diff
@@ -1032,8 +1042,8 @@ impl TendermintExtension {
             let view = t.view.load(AtomicOrdering::SeqCst);
             let du = Duration::seconds(view as i64);
 
-            api.clear_timer(ENGINE_TIMEOUT_TOKEN).expect("Timer clear succeeds");
-            api.set_timer_once(ENGINE_TIMEOUT_TOKEN, self.timeouts.timeout(&step) + du).expect("Timer set succeeds");
+            api.clear_timer(ENGINE_TIMEOUT_TOKEN + counter).expect("Timer clear succeeds");
+            api.set_timer_once(ENGINE_TIMEOUT_TOKEN + counter, self.timeouts.timeout(&step) + du).expect("Timer set succeeds");
```

Comment:
You write code that `clear_timer` and `set_timer_once` receives the same counter value.
But should we clear the timer with the previous counter and set new timer with the new value?

Maybe the code should be like below.

```
api.clear_timer(ENGINE_TIMEOUT_TOKEN + counter - 1).expect("Timer clear succeeds");
api.set_timer_once(ENGINE_TIMEOUT_TOKEN + counter, self.timeouts.timeout(&step) + du).expect("Timer set succeeds");
```

Reply:
Oh, I've missed this. I'll fix this.